### PR TITLE
Feature - better status - shows owners and last dataset ingestion  

### PR DESCRIPTION
**Pull request by iakov-aws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cid/helpers/quicksight/__init__.py [hunk 1]
```diff
@@ -906,6 +897,33 @@ def describe_dataset(self, id, timeout: int=1) -> Dataset:
...
+            return '<RED>NotFound<END>'
+        except self.client.exceptions.AccessDeniedException:
+            return '<YELLOW>AccessDenied<END>'
+        if not ingestions:
+            return None
+        last_ingestion = ingestions[0] # Suppose it is the latest
+        status = last_ingestion.get('IngestionStatus')
+        time_ago = ago(last_ingestion.get('CreatedTime'))
+        if last_ingestion.get('ErrorInfo', {}).get('Type') == "DATA_SET_NOT_SPICE":
+            return '<BLUE>DIRECT_QUERY<END>'
```
Comment (vmindru) [inline cid/helpers/quicksight/__init__.py:917]: why not also define status here, and return it at line 925  return f'{status} {time_ago}'  ? it's the same login as in UI time ago is the creationt ime.

Comment (vmindru) [reply cid/helpers/quicksight/__init__.py:917]: I mean nothing ciriminal if statys liek this only  helps with code consistenncy

File: cid/helpers/quicksight/dashboard.py [hunk 2]
```diff
@@ -100,53 +101,59 @@ def status(self) -> str:
...
-            logger.debug(f"TemplateVersion  {str(self.deployed_version): <9} -->  {str(self.latest_version): <6}")
-            
+            cid_print(f"  <BOLD>Version:<END>   <YELLOW>{str(cid_version): <8} --> {str(cid_version_latest): <8}<END>")
+            cid_print(f"  <BOLD>VersionId:<END> <YELLOW>{str(self.deployed_version): <8} --> {str(self.latest_version): <8}<END>")
+
+        cid_print('  <BOLD>Owners:<END>')
+        try:
+            permissions = self.qs.get_dashboard_permissions(self.id)
+            for permission in permissions:
+                if 'quicksight:UpdateDashboardPermissions' in permission["Actions"]:
```
Comment (vmindru) [inline cid/helpers/quicksight/dashboard.py:141]: Does it really require UpdateDashboardPermissions?

File: cid/helpers/quicksight/dashboard.py [hunk 3]
```diff
@@ -100,53 +101,59 @@ def status(self) -> str:
...
+                    cid_print('    ' + permission["Principal"].split('user/default/')[-1])
+        except Exception as exc:
+            if "AccessDenied" in str(exc):
+                cid_print('     <RED>AccessDenied<END>')
```
Comment (vmindru) [inline cid/helpers/quicksight/dashboard.py:145]: If comment at 141 is vali myabe add here which permission exactly is expected, else this error might be a bit more to generic.

File: cid/helpers/quicksight/__init__.py (hunk 1 above)
Comment (iakov-aws) [reply cid/helpers/quicksight/__init__.py:917]: If dataset is DATA_SET_NOT_SPICE the status can be empty

File: cid/helpers/quicksight/dashboard.py (hunk 3 above)
Comment (iakov-aws) [reply cid/helpers/quicksight/dashboard.py:145]: This access denied can accure when reading. The L141 just checks permissions of user on the dashboard

**Review by vmindru — APPROVED** [review]

**Review by yprikhodko — APPROVED** [review]

**Pull request merged**